apps/web/components/v2/eventtype/EventLimitsTab.tsx

```diff
@@ -1,21 +1,31 @@
 import { useAutoAnimate } from "@formkit/auto-animate/react";
 import * as RadioGroup from "@radix-ui/react-radio-group";
 import { EventTypeSetupInfered, FormValues } from "pages/event-types/[type]";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { useFormContext, Controller, useWatch } from "react-hook-form";
 
 import { classNames } from "@calcom/lib";
+import convertToNewDurationType from "@calcom/lib/convertToNewDurationType";
+import findDurationType from "@calcom/lib/findDurationType";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { PeriodType } from "@calcom/prisma/client";
 import type { BookingLimit } from "@calcom/types/Calendar";
 import { Icon } from "@calcom/ui";
-import { Select, Switch, Label, Input, MinutesField, Button, SettingsToggle } from "@calcom/ui/v2";
+import { Select, Switch, Label, Input, Button, SettingsToggle, InputField } from "@calcom/ui/v2";
 import DateRangePicker from "@calcom/ui/v2/core/form/date-range-picker/DateRangePicker";
 
+export const durationArray: string[] = ["minutes"];
```

Comment:
I didn't notice the usage of this array outside of the component before. I think accessing changing variables outside the render function is very prone to breaking. I would suggest adding a `useRef` variable inside your component where you track it's previous value. In that case you can also use a string instead of an array by eg setting `previousValueRef.current = ....`. In my opinion that also makes it again a bit more readable 🥰

Reply:
@JeroenReumkens If I use a `useRef` how would I access the value of the ref outside of this component? I need to use the current duration type to convert the `minimumBookingNotice` to minutes before sending it to prisma.

Reply:
@JeroenReumkens I'm not very familiar with using refs, but I will try and figure this out! Thank you for the suggestion!